iothub/device/src/InternalClient.cs

```diff
@@ -999,6 +999,9 @@ internal async Task OnMethodCalledAsync(MethodRequestInternal methodRequestInter
                 methodResponseInternal = new MethodResponseInternal(methodRequestInternal.RequestId, (int)MethodResponseStatusCode.BadRequest);
 
                 await SendMethodResponseAsync(methodResponseInternal, methodRequestInternal.CancellationToken).ConfigureAwait(false);
+
+                methodResponseInternal?.Dispose();
```

Comment:
Won't this already happen on 1016 in the finally clause?

Reply:
Correct. Fixed.